hr_public_holidays/models/hr_public_holidays.py

```diff
@@ -33,11 +35,55 @@ class HrPublicHolidays(models.Model):
         'Country'
     )
 
+    @api.model
+    def create(self, values):
+        holiday = super(HrPublicHolidays, self).create(values)
+        for line in holiday.line_ids:
+            leave = self.env['resource.calendar.leaves'].create(
+                {'name': line.name,
+                 'date_from': datetime.datetime.combine(
+                     fields.Datetime.from_string(line.date),
+                     datetime.time.min),
+                 'date_to': datetime.datetime.combine(
+                     fields.Datetime.from_string(line.date),
+                     datetime.time.max)})
+            line.calendar_leave_id = leave.id
+        return holiday
+
+    @api.multi
+    def write(self, vals):
+        holiday = super(HrPublicHolidays, self).write(vals)
+        for line in holiday.line_ids:
+            line.calendar_leave_id.unlink()
+            leave = self.env['resource.calendar.leaves'].create(
+                {'name': line.name,
+                 'date_from': datetime.datetime.combine(
+                     fields.Datetime.from_string(line.date),
+                     datetime.time.min),
+                 'date_to': datetime.datetime.combine(
+                     fields.Datetime.from_string(line.date),
+                     datetime.time.max)})
+            line.calendar_leave_id = leave.id
+        return holiday
+
+    @api.onchange('line_ids')
+    def _onchange_line_ids(self):
+        for line in self.line_ids:
+            leave = self.env['resource.calendar.leaves'].create(
```

Comment:
I don't like the idea to use a create() inside an onchange method. You may want to adopt the proposal to use a workflow, this way you don't need this onchange method at all.